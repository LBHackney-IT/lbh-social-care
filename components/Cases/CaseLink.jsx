import PropTypes from 'prop-types';
import Link from 'next/link';
import * as form from 'data/forms';

const getLink = (recordId, { form_name_overall, ...caseFormData }) => {
  const fileName = form[form_name_overall] ? form_name_overall : null;

  switch (form_name_overall) {
    case 'API_Allocation':
    case 'API_Deallocation':
      return `/people/${caseFormData.mosaic_id}/allocations/${caseFormData.allocation_id}?recordId=${recordId}`;
    case fileName:
      return `/people/${caseFormData.mosaic_id}/records/${recordId}`;
    default:
      return null;
  }
};

const CaseLink = ({ recordId, externalUrl, caseFormData }) => {
  if (externalUrl) {
    return (
      <a
        href={externalUrl}
        target="_blank"
        rel="noreferrer noopener"
        className="govuk-link"
      >
        View
      </a>
    );
  }
  const internalLink = getLink(recordId, caseFormData);
  return internalLink ? <Link href={internalLink}>View</Link> : null;
};

CaseLink.propTypes = {
  recordId: PropTypes.string.isRequired,
  externalUrl: PropTypes.string,
  caseFormData: PropTypes.shape({
    form_name_overall: PropTypes.string.isRequired,
  }).isRequired,
};

export default CaseLink;
