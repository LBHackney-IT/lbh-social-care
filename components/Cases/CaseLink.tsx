import Link from 'next/link';
import forms from 'data/flexibleForms';
import * as form from 'data/forms';

import {
  AllocationCaseFormData,
  CaseFormData,
  DeallocationCaseFormData,
  WarningNoteCaseFormData,
} from 'types';

const getWarningNoteDetailsPageName = (form_name: string) => {
  switch (form_name) {
    case 'Warning Note Created':
      return 'note-created';
    case 'Warning Note Reviewed':
      return 'note-reviews';
    case 'Warning Note Ended':
      return 'note-end';
    default:
      return null;
  }
};

const getLink = (
  recordId: string,
  { form_name_overall, ...caseFormData }: CaseFormData
) => {
  const fileName = (form as Record<string, unknown>)[form_name_overall]
    ? form_name_overall
    : null;

  switch (form_name_overall) {
    case 'API_Allocation':
    case 'API_Deallocation':
      return `/people/${caseFormData.mosaic_id}/allocations/${
        (caseFormData as AllocationCaseFormData | DeallocationCaseFormData)
          .allocation_id
      }?recordId=${recordId}`;
    case fileName:
      return `/people/${caseFormData.mosaic_id}/records/${recordId}`;
    case 'Historical_Case_Note':
      return `/people/${caseFormData.mosaic_id}/records/${recordId}?is_historical=${caseFormData.is_historical}`;
    case 'Historical_Visit':
      return `/people/${caseFormData.mosaic_id}/visits/${recordId}?is_historical=${caseFormData.is_historical}`;
    case 'API_WarningNote':
      return `/people/${caseFormData.mosaic_id}/warning-notes/${
        (caseFormData as WarningNoteCaseFormData).warning_note_id
      }/view/${getWarningNoteDetailsPageName(caseFormData.form_name)}`;
    default:
      return null;
  }
};

interface Props {
  recordId: string;
  externalUrl?: string;
  caseFormData: CaseFormData;
  formName: string;
  personId: number;
  children?: React.ReactChild;
}

const CaseLink = ({
  formName,
  externalUrl,
  caseFormData,
  recordId,
  personId,
  children,
}: Props): React.ReactElement => {
  const form = formName ? forms?.find((form) => form.id === formName) : false;

  if (form)
    return (
      <Link href={`/people/${personId}/submissions/${recordId}`}>
        <a className="lbh-link">{form?.name || 'View'}</a>
      </Link>
    );

  if (externalUrl) {
    return (
      <a
        href={externalUrl}
        target="_blank"
        rel="noreferrer noopener"
        className="lbh-link"
      >
        View
      </a>
    );
  }
  const internalLink = getLink(recordId, caseFormData);
  return internalLink ? (
    <Link href={internalLink}>
      <a className="lbh-link">{children || 'View'}</a>
    </Link>
  ) : (
    <>{formName}</>
  );
};

export default CaseLink;
