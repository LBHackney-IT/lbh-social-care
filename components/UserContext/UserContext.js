import { createContext, useContext, useEffect, useState } from 'react';
import { useRouter } from 'next/router';

import { AUTH_WHITELIST } from 'utils/auth';
import { getUser } from 'utils/api/checkAuth';
import Layout from 'components/Layout';
import Spinner from 'components/Spinner/Spinner';

export const UserContext = createContext({
  user: undefined,
  setUser: () => {},
});

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState();
  const [redirect, setRedirect] = useState();
  const [isLoading, setLoading] = useState(true);
  const router = useRouter();
  useEffect(() => {
    const initializeAuth = async () => {
      try {
        const data = getUser();
        setUser(data);
      } catch (e) {
        setUser();
        setRedirect(e.response.status === 403 ? '/access-denied' : '/login');
      }
      setLoading(false);
    };
    initializeAuth();
  }, []);
  if (isLoading || !router.isReady) {
    return (
      <Layout>
        <Spinner />
      </Layout>
    );
  }
  if (
    typeof window !== 'undefined' &&
    AUTH_WHITELIST.includes(router.pathname) === Boolean(user)
  ) {
    router.push(redirect || location);
    return <></>;
  }
  return (
    <UserContext.Provider
      value={{
        user,
        isLoading,
        setUser,
        redirect,
      }}
    >
      {children}
    </UserContext.Provider>
  );
};

export function useAuth() {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
}
