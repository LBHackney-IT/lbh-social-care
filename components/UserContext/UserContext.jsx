import { createContext, useContext } from 'react';
import { useRouter } from 'next/router';

import { shouldRedirect } from 'utils/auth';
import { isBrowser } from 'utils/ssr';

export const UserContext = createContext({
  user: undefined,
  setUser: () => {},
});

export const AuthProvider = ({ children, user }) => {
  const { pathname, push } = useRouter();
  const redirect = isBrowser() && shouldRedirect(pathname, user);
  if (redirect) {
    push(redirect);
    return <></>;
  }
  return (
    <UserContext.Provider
      value={{
        user,
      }}
    >
      {children}
    </UserContext.Provider>
  );
};

export function useAuth() {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
}
