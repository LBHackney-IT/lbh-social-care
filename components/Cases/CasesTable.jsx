import PropTypes from 'prop-types';
import { formatDate } from 'utils/date';

import CaseLink from './CaseLink';

const CasesEntry = ({
  recordId,
  formName,
  caseFormUrl,
  officerEmail,
  dateOfEvent,
  caseFormData,
  caseFormTimestamp,
}) => (
  <tr className="govuk-table__row">
    <td className="govuk-table__cell govuk--timestamp">
      {formatDate(dateOfEvent || caseFormTimestamp)}{' '}
    </td>
    <td className="govuk-table__cell">{formName}</td>
    <td className="govuk-table__cell">
      {officerEmail && `- created by ${officerEmail}`}
    </td>
    <td className="govuk-table__cell govuk-button--secondary'">
      <CaseLink
        recordId={recordId}
        externalUrl={caseFormUrl}
        caseFormData={caseFormData}
      />
    </td>
  </tr>
);

CasesEntry.propTypes = {
  recordId: PropTypes.string.isRequired,
  formName: PropTypes.string.isRequired,
  caseFormUrl: PropTypes.string.isRequired,
  officerEmail: PropTypes.string.isRequired,
  dateOfEvent: PropTypes.string,
  caseFormTimestamp: PropTypes.string,
  caseFormData: PropTypes.object.isRequired,
};

const CasesTable = ({ records }) => (
  <table className="govuk-table">
    <tbody className="govuk-table__body">
      {records.map((result) => (
        <CasesEntry key={result.recordId} {...result} />
      ))}
    </tbody>
  </table>
);

CasesTable.propTypes = {
  records: PropTypes.arrayOf(PropTypes.shape(CasesEntry.propTypes)).isRequired,
};

export default CasesTable;
