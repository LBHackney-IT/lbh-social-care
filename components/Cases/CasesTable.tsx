import { Case } from 'types';
import { formatDate, isDateValid } from 'utils/date';

import CaseLink from './CaseLink';

const CaseNotePersonId = ({ personId }: Case) => (
  <td key="personId" className="govuk-table__cell">
    {personId}
  </td>
);

const CaseNoteDate = ({ dateOfEvent, caseFormTimestamp }: Case) => (
  <td
    key="date"
    className="govuk-table__cell govuk--timestamp"
    style={{ width: '120px' }}
  >
    {(dateOfEvent && formatDate(dateOfEvent)) ||
      (caseFormTimestamp && formatDate(caseFormTimestamp))}
  </td>
);

const CaseNoteTitle = ({ formName, caseFormData }: Case) => (
  <td key="title" className="govuk-table__cell">
    {
      <>
        {['ASC_case_note', 'CFS_case_note'].includes(
          caseFormData.form_name_overall
        ) && 'Case Note - '}
        {formName}
        {caseFormData.case_note_title && (
          <div>{caseFormData.case_note_title}</div>
        )}
      </>
    }
  </td>
);

const CaseNoteOfficer = ({ officerEmail }: Case) => (
  <td key="officer" className="govuk-table__cell">
    {officerEmail}
  </td>
);

const CaseNoteAction = ({ recordId, caseFormUrl, caseFormData }: Case) => (
  <td
    key="action"
    className="govuk-table__cell govuk-table__cell--numeric"
    style={{ width: '50px', textAlign: 'center' }}
  >
    <CaseLink
      recordId={recordId}
      externalUrl={caseFormUrl}
      caseFormData={caseFormData}
    />
  </td>
);

const CaseNoteName = ({ firstName, lastName }: Case) => (
  <td key="name" className="govuk-table__cell">
    {firstName} {lastName}
  </td>
);

const CaseNoteBirthday = ({ dateOfBirth }: Case) => (
  <td key="birthday" className="govuk-table__cell">
    {dateOfBirth && isDateValid(dateOfBirth) && dateOfBirth}
  </td>
);

const tableEntities = {
  person_id: { text: 'Person id', component: CaseNotePersonId },
  first_name: { text: 'Client Name', component: CaseNoteName },
  date_of_birth: { text: 'Date of birth', component: CaseNoteBirthday },
  date_of_event: { text: 'Date created', component: CaseNoteDate },
  formName: { text: 'Record type', component: CaseNoteTitle },
  officer_email: { text: 'Created by', component: CaseNoteOfficer },
  action: { text: 'Action', component: CaseNoteAction },
};

export type CaseTableColumns = keyof typeof tableEntities;

interface Props {
  records: Case[];
  columns: CaseTableColumns[];
  sort?: {
    sort_by?: string;
    order_by?: string;
  };
  onSort?: (id: string) => void;
}

const CasesTable = ({
  records,
  columns,
  sort = {},
  onSort,
}: Props): React.ReactElement => (
  <table className="govuk-table">
    <thead className="govuk-table__head">
      <tr className="govuk-table__row">
        {columns.map((column) => (
          <th
            key={column}
            scope="col"
            className="govuk-table__header"
            role={onSort && 'button'}
            onClick={() => onSort && onSort(column)}
          >
            {tableEntities[column].text}{' '}
            {column === sort.sort_by && (
              <>{sort.order_by === 'desc' ? 'ðŸ”½' : 'ðŸ”¼'}</>
            )}
          </th>
        ))}
      </tr>
    </thead>
    <tbody className="govuk-table__body">
      {records.map((record) => (
        <tr className="govuk-table__row" key={record.recordId}>
          {columns.map((column) => tableEntities[column].component(record))}
        </tr>
      ))}
    </tbody>
  </table>
);

export default CasesTable;
