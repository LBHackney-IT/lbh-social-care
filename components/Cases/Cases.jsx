import PropTypes from 'prop-types';

import CasesTable from 'components/Cases/CasesTable';
import ErrorMessage from 'components/ErrorMessage/ErrorMessage';
import ErrorSummary from 'components/ErrorSummary/ErrorSummary';
import Button from 'components/Button/Button';
import Spinner from 'components/Spinner/Spinner';
import { useAuth } from 'components/UserContext/UserContext';
import { getCasesByResident } from 'utils/api/cases';

const Cases = ({ id }) => {
  const { data, size, setSize, error } = getCasesByResident(id);
  const results = data?.length > 0 && {
    cases: data.reduce((acc, { cases }) => [...acc, ...cases], []),
    nextCursor: data[data.length - 1].nextCursor,
  };
  if (error) {
    return <ErrorMessage />;
  }
  if (!results) {
    return (
      <div>
        <Spinner />
      </div>
    );
  }
  return (
    <>
      {results?.cases.length > 0 ? (
        <CasesTable records={results.cases} />
      ) : (
        <p className="govuk-body govuk-!-margin-top-5">Records not found</p>
      )}
      <div style={{ height: '50px', textAlign: 'center' }}>
        {size > data.length ? (
          <Spinner />
        ) : (
          results?.nextCursor && (
            <Button label="load more" onClick={() => setSize(size + 1)} />
          )
        )}
      </div>
    </>
  );
};

const CasesWrapper = ({ id, person }) => {
  const { user } = useAuth();
  return (
    <>
      <div className="lbh-table-header">
        <div>
          <h3 className="govuk-fieldset__legend--m govuk-custom-text-color govuk-!-margin-top-0">
            RECORDS HISTORY
          </h3>
          <p className="govuk-label  govuk-!-margin-top-0">
            Linked files are read only
          </p>
        </div>
        <Button label="Add a new record" route={`${id}/records`} />
      </div>
      <hr className="govuk-divider" />
      {user.hasUnrestrictedPermissions || !person.restricted ? (
        <Cases id={id} />
      ) : (
        <ErrorSummary
          title="RESTRICTED"
          body="The records for this profile are restricted for viewing"
        />
      )}
    </>
  );
};

CasesWrapper.propTypes = {
  id: PropTypes.string.isRequired,
  person: PropTypes.shape({
    restricted: PropTypes.bool,
  }).isRequired,
};

export default CasesWrapper;
