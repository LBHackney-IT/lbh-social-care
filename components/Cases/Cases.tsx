import CasesTable from 'components/Cases/CasesTable';
import ErrorMessage from 'components/ErrorMessage/ErrorMessage';
import ErrorSummary from 'components/ErrorSummary/ErrorSummary';
import Button from 'components/Button/Button';
import Spinner from 'components/Spinner/Spinner';
import { useAuth } from 'components/UserContext/UserContext';
import { useCasesByResident } from 'utils/api/cases';

import { Case, Resident, User } from 'types';
import { canManageCases } from '../../lib/permissions';

interface Props {
  id: number;
}

const Cases = ({ id }: Props): React.ReactElement => {
  const { data, size, setSize, error } = useCasesByResident(id);
  const results =
    data && data.length > 0
      ? {
          cases: data.reduce<Case[]>(
            (acc, { cases }) => [...acc, ...cases],
            []
          ),
          nextCursor: data[data.length - 1].nextCursor,
        }
      : null;
  if (error) {
    return <ErrorMessage />;
  }
  if (!data || !results) {
    return (
      <div>
        <Spinner />
      </div>
    );
  }
  return (
    <>
      {results.cases.length > 0 ? (
        <CasesTable
          records={results.cases}
          columns={['date_of_event', 'formName', 'officer_email', 'action']}
        />
      ) : (
        <p className="govuk-body govuk-!-margin-top-5">Records not found</p>
      )}
      <div style={{ height: '50px', textAlign: 'center' }}>
        {size > data.length ? (
          <Spinner />
        ) : (
          results.nextCursor && (
            <Button label="load more" onClick={() => setSize(size + 1)} />
          )
        )}
      </div>
    </>
  );
};

interface WrapperProps {
  id: number;
  person: Resident;
}

const CasesWrapper = ({ id, person }: WrapperProps): React.ReactElement => {
  const { user } = useAuth() as { user: User };

  const userCanManageCases = canManageCases(user, person);

  return (
    <div>
      <div className="lbh-table-header">
        <div>
          <h3 className="govuk-fieldset__legend--m govuk-custom-text-color">
            RECORDS HISTORY
          </h3>
          <span className="govuk-body">Linked files are read only</span>
        </div>
        {userCanManageCases && (
          <Button label="Add a new record" route={`${id}/records`} />
        )}
      </div>
      <hr className="govuk-divider" />
      {userCanManageCases ? (
        <Cases id={id} />
      ) : (
        <ErrorSummary
          title="RESTRICTED"
          body="Some details for this person are restricted due to your permissions."
        />
      )}
    </div>
  );
};

export default CasesWrapper;
