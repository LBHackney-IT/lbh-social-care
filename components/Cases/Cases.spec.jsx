import { render, waitFor, fireEvent } from '@testing-library/react';

import { useCasesByResident } from 'utils/api/cases';
import { UserContext } from 'components/UserContext/UserContext';

import Cases from './Cases';

jest.mock('utils/api/cases', () => ({
  useCasesByResident: jest.fn(),
}));

describe('Cases component', () => {
  const props = {
    id: '44000000',
    person: {
      firstName: 'Foo',
      lastName: 'Bar',
      mosaicId: '123',
      ageContext: 'A',
    },
  };

  it('should render records properly', async () => {
    const mockSetSize = jest.fn();
    useCasesByResident.mockImplementation(() => ({
      size: 2,
      setSize: mockSetSize,
      data: [
        {
          cases: [
            {
              caseFormTimestamp: '2020-12-04',
              dateOfBirth: '2020-12-13',
              firstName: 'foo',
              lastName: 'FOO',
              personId: '44000001',
              formName: 'Foo_foo',
              recordId: '12347',
              caseFormUrl: 'http//foo/foo',
              officerEmail: 'foo@foo.co.uk',
              dateOfEvent: '2020-12-23',
              caseFormData: { form_name_overall: 'foo' },
            },
          ],
          nextCursor: 1,
        },
        {
          cases: [
            {
              caseFormTimestamp: '2020-12-04',
              dateOfBirth: '2020-12-13',
              firstName: 'bar',
              lastName: 'BAR',
              personId: '44000001',
              formName: 'Bar_bar',
              recordId: '12345',
              caseFormUrl: 'http//bar/bar',
              officerEmail: 'bar@bar.co.uk',
              caseFormData: { form_name_overall: 'foo' },
            },
          ],
          nextCursor: 2,
        },
      ],
    }));
    const { asFragment, getByRole, getAllByText } = render(
      <UserContext.Provider
        value={{
          user: { hasAdminPermissions: true },
        }}
      >
        <Cases {...props} />
      </UserContext.Provider>
    );
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
    const button = getByRole('button', { name: 'load more' });
    expect(getAllByText('View')).toHaveLength(2);
    fireEvent.click(button);
    expect(mockSetSize).toHaveBeenCalled();
    expect(mockSetSize).toHaveBeenCalledWith(3);
  });

  it('should render no records ', async () => {
    useCasesByResident.mockImplementation(() => ({
      size: 0,
      data: [
        {
          cases: [],
          nextCursor: 1,
        },
      ],
    }));
    const { asFragment, getByText } = render(
      <UserContext.Provider
        value={{
          user: { hasAdminPermissions: true },
        }}
      >
        <Cases {...props} />
      </UserContext.Provider>
    );
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
    const label = getByText('Records not found');
    expect(label).toBeInTheDocument();
  });

  it('should render a error message when a person is restricted', async () => {
    const props = {
      id: '44000000',
      person: {
        restricted: true,
      },
    };
    const { asFragment, getByText } = render(
      <UserContext.Provider
        value={{
          user: { hasAdminPermissions: true },
        }}
      >
        <Cases {...props} />
      </UserContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(useCasesByResident).not.toHaveBeenCalled();
    const title = getByText('RESTRICTED');
    expect(title).toBeInTheDocument();
  });
});
