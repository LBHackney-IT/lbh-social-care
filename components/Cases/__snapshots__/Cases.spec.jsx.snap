// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cases component should render no records  1`] = `
<DocumentFragment>
  <div
    class="lbh-table-header"
  >
    <div>
      <h3
        class="govuk-fieldset__legend--m govuk-custom-text-color govuk-!-margin-top-0"
      >
        RECORDS HISTORY
      </h3>
      <p
        class="govuk-label  govuk-!-margin-top-0"
      >
        Linked files are read only
      </p>
    </div>
    <button
      class="govuk-button"
      data-module="govuk-button"
    >
      Add a new record
    </button>
  </div>
  <hr
    class="govuk-divider"
  />
  <div
    class="lbh-table-header"
  >
    <div>
      <h3
        class="govuk-fieldset__legend--m govuk-custom-text-color govuk-!-margin-top-0"
      >
        RECORDS HISTORY
      </h3>
      <p
        class="govuk-label  govuk-!-margin-top-0"
      >
        Linked files are read only
      </p>
    </div>
    <button
      class="govuk-button"
      data-module="govuk-button"
    >
      Add a new record
    </button>
  </div>
  <hr
    class="govuk-divider"
  />
  <p
    class="govuk-body govuk-!-margin-top-5"
  >
    Records not found
  </p>
  <div
    style="height: 50px; text-align: center;"
  >
    <button
      class="govuk-button"
      data-module="govuk-button"
    >
      load more
    </button>
  </div>
</DocumentFragment>
`;

exports[`Cases component should render records properly 1`] = `
<DocumentFragment>
  <div
    class="lbh-table-header"
  >
    <div>
      <h3
        class="govuk-fieldset__legend--m govuk-custom-text-color govuk-!-margin-top-0"
      >
        RECORDS HISTORY
      </h3>
      <p
        class="govuk-label  govuk-!-margin-top-0"
      >
        Linked files are read only
      </p>
    </div>
    <button
      class="govuk-button"
      data-module="govuk-button"
    >
      Add a new record
    </button>
  </div>
  <hr
    class="govuk-divider"
  />
  <div
    class="lbh-table-header"
  >
    <div>
      <h3
        class="govuk-fieldset__legend--m govuk-custom-text-color govuk-!-margin-top-0"
      >
        RECORDS HISTORY
      </h3>
      <p
        class="govuk-label  govuk-!-margin-top-0"
      >
        Linked files are read only
      </p>
    </div>
    <button
      class="govuk-button"
      data-module="govuk-button"
    >
      Add a new record
    </button>
  </div>
  <hr
    class="govuk-divider"
  />
  <table
    class="govuk-table"
  >
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
        >
          Jun 11, 1929 
        </td>
        <td
          class="govuk-table__cell"
        >
          Foo_foo
        </td>
        <td
          class="govuk-table__cell"
        >
          - created by foo@foo.co.uk
        </td>
        <td
          class="govuk-table__cell govuk-button--secondary'"
        >
          <a
            class="govuk-link"
            href="http//foo/foo"
            rel="noopener"
            target="_blank"
          >
            View
          </a>
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell govuk--timestamp"
        >
          Jun 11, 1929 
        </td>
        <td
          class="govuk-table__cell"
        >
          Bar_bar
        </td>
        <td
          class="govuk-table__cell"
        >
          - created by bar@bar.co.uk
        </td>
        <td
          class="govuk-table__cell govuk-button--secondary'"
        >
          <a
            class="govuk-link"
            href="http//bar/bar"
            rel="noopener"
            target="_blank"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    style="height: 50px; text-align: center;"
  >
    <button
      class="govuk-button"
      data-module="govuk-button"
    >
      load more
    </button>
  </div>
</DocumentFragment>
`;
